core/modules/editor/factory.js

```diff
@@ -205,6 +216,9 @@ function editTextWidgetFactory(toolbarEngine,nonToolbarEngine) {
 		// Get the rest of our parameters
 		this.editTag = this.getAttribute("tag",tag) || "input";
 		this.editType = this.getAttribute("type",type);
+		// Fetch type module if any exists.
+		// This module takes kare of converting between TW5 and HTML5 representation of the value.
+		this.typeModule = EditTextWidget.typeModules[this.editType];
```

Comment:
I don't think any other widgets instantiate submodules, but the pattern we use elsewhere is like this:

https://github.com/Jermolene/TiddlyWiki5/blob/54cfda76ee353190f1cf0210b9071894fb1a5690/core/modules/parsers/wikiparser/wikiparser.js#L62-L65

In otherwords, we avoid calling `$tw.modules.*` at parse time, we do it when the object is instantiated.